### PR TITLE
updated source code to the latest CF

### PR DESCRIPTION
**Pull request by txiang61**

### CODE REVIEW

File: src/units/UnitsAnnotatedTypeFactory.java [hunk 1]
```diff
@@ -369,6 +370,12 @@ public UnitsTypecheckImplicitsTreeAnnotator() {
         public UnitsPropagationTreeAnnotator() {
             super(UnitsAnnotatedTypeFactory.this);
         }
+        
+        @Override
+        public Void visitLiteral(LiteralTree tree, AnnotatedTypeMirror type) {
+        	
+        	return super.visitLiteral(tree, type);
```
Comment (wmdietl) [inline src/units/UnitsAnnotatedTypeFactory.java]: Why add this override if it only calls super unchanged?

File: src/units/UnitsAnnotatedTypeFactory.java [hunk 2]
```diff
@@ -433,6 +440,24 @@ public Void visitBinary(BinaryTree binaryTree, AnnotatedTypeMirror type) {
...
     }
+    
+    @Override
+    protected TypeAnnotator createTypeAnnotator() {
+        return new ListTypeAnnotator(
+                new UnitsDefaultForTypeAnnotator(this), super.createTypeAnnotator());
+    }
+    
+    protected class UnitsDefaultForTypeAnnotator extends DefaultForTypeAnnotator {
+        // Programmatically set the qualifier
```
Comment (wmdietl) [inline src/units/UnitsAnnotatedTypeFactory.java]: Do you understand why these aren't just set declaratively at
https://github.com/opprop/units-inference/blob/master/src/units/qual/Dimensionless.java#L15 ?

File: src/units/UnitsLiteralTreeAnnotator.java [hunk 3]
```diff
@@ -2,25 +2,25 @@
...
     UnitsRepresentationUtils unitsRepUtils = UnitsRepresentationUtils.getInstance();
 
-    // Programmatically set the qualifier implicits
-    public UnitsImplicitsTreeAnnotator(AnnotatedTypeFactory atf) {
+    // Programmatically set the qualifier for literals
+    public UnitsLiteralTreeAnnotator(AnnotatedTypeFactory atf) {
         super(atf);
 
-        // set BOTTOM as the implicit qualifier for null literals
+        // set BOTTOM as the literal qualifier for null literals
```
Comment (wmdietl) [inline src/units/UnitsLiteralTreeAnnotator.java:17]: Couldn't these also be done declaratively?

Comment (txiang61) [reply src/units/UnitsLiteralTreeAnnotator.java:17]: The type checking mode and the inference mode have different literals defaults.

File: src/units/UnitsAnnotatedTypeFactory.java (hunk 2 above)
Comment (txiang61) [reply src/units/UnitsAnnotatedTypeFactory.java]: I believe this is because these defaults are only set for type checking mode and not inference mode.

**Pull request closed without merging**